Detections/MultipleDataSources/SuspiciousModificationofGlobalAdminProperties.yaml

```diff
@@ -11,36 +11,44 @@ requiredDataConnectors:
+queryPeriod: 14d
 triggerOperator: gt
 triggerThreshold: 0
 tactics:
   - PrivilegeEscalation
 relevantTechniques:
   - T1078.004
-tags:
-  - GuestorExternalIdentities
 query: |
-  let lookback = 1d;
-  let GlobalAdmins = IdentityInfo
-  | where TimeGenerated > ago(lookback)
-  | extend IsGlobalAdmin = set_has_element(AssignedRoles, "Global Administrator")
-  | where IsGlobalAdmin == true
-  | distinct AccountUPN;
-  GlobalAdmins
-  | join kind=inner ( AuditLogs
-  | where TimeGenerated > ago(lookback)
-  | where OperationName=~ "Update user" 
-  | where Result =~ "success" 
-  | mv-expand TargetResources 
-  | mv-expand TargetResources.modifiedProperties 
-  | extend displayName_ = tostring(TargetResources_modifiedProperties.displayName) , oldValue_ = tostring(TargetResources_modifiedProperties.oldValue), newValue_ = tostring(TargetResources_modifiedProperties.newValue)
-  | where displayName_ == "UserPrincipalName" and oldValue_ !has "#EXT" and newValue_ has "#EXT"
-  | extend InitiatingApp = tostring(parse_json(tostring(InitiatedBy.app)).displayName) 
-  | extend Initiator = iif(isnotempty(InitiatingApp), InitiatingApp, tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName)) , IPAddress = tostring(InitiatedBy.["user"].["ipAddress"])
-  ) on $left.AccountUPN == $right.oldValue_
-  | project TimeGenerated, AADTenantId, IPAddress, Initiator, displayName_, oldValue_, newValue_
+  let query_frequency = 1h;
+  let query_period = 14d;
+  IdentityInfo
+  | where TimeGenerated > ago(query_period)
+  | where set_has_element(AssignedRoles, "Global Administrator")
+  | distinct AccountUPN, AccountObjectId
+  | join kind=inner (
+      AuditLogs
+      | where TimeGenerated > ago(query_frequency)
+      | where OperationName=~ "Update user" and Result =~ "success"
+      // | where isnotempty(InitiatedBy["user"])
+      | mv-expand TargetResource = TargetResources
+      | where TargetResource["type"] == "User"
+      | extend AccountObjectId = tostring(TargetResource["id"])
+      | where not(tostring(TargetResource["modifiedProperties"]) == "[]")
+      | mv-apply modifiedProperty = TargetResource["modifiedProperties"] on (
+          summarize modifiedProperties = make_bag(
+              bag_pack(tostring(modifiedProperty["displayName"]),
+                  bag_pack("oldValue", trim(@'[\"\s]+', tostring(modifiedProperty["oldValue"])),
+                      "newValue", trim(@'[\"\s]+', tostring(modifiedProperty["newValue"])))))
+      )
+      | where not(tostring(modifiedProperties["Included Updated Properties"]["newValue"]) in ("LastDirSyncTime", ""))
+      | where not(isnotempty(modifiedProperties["StrongAuthenticationPhoneAppDetail"]) and tostring(array_sort_asc(extract_all(@'\"Id\"\:\"([^\"]+)\"', tostring(modifiedProperties["StrongAuthenticationPhoneAppDetail"]["newValue"])))) == tostring(array_sort_asc(extract_all(@'\"Id\"\:\"([^\"]+)\"', tostring(modifiedProperties["StrongAuthenticationPhoneAppDetail"]["oldValue"])))))
```

Comment:
@devikamehra the proposed condition is not equivalent to the PR condition.

In the proposed condition, if a event does NOT have StrongAuthenticationPhoneAppDetail, that event won't appear in the results (undesired behaviour because other properties can be modified).

The PR condition tries to exclude events where (if StrongAuthenticationPhoneAppDetail is populated) the StrongAuthenticationPhoneAppDetail has the same devices after the modification (thus only the LastTimeUsed field is updated).

The proposed condition to be equivalent to PR condition it should be:

| where isempty(modifiedProperties["StrongAuthenticationPhoneAppDetail"]) or tostring(array_sort_asc(extract_all(@'"Id":"([^\"]+)"', tostring(modifiedProperties["StrongAuthenticationPhoneAppDetail"]["newValue"])))) != tostring(array_sort_asc(extract_all(@'"Id":"([^\"]+)"', tostring(modifiedProperties["StrongAuthenticationPhoneAppDetail"]["oldValue"]))))

And I believe this is not easier to read than the PR condition.

Moreover, ```not(x == y)``` and ```x != y``` do not have the same behaviour when one of the parameters is empty.